express-mutations/server.ts

```diff
@@ -0,0 +1,114 @@
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/pagila',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+app.use(express.json());
+
+// Endpoint for testing
+app.get('/api/actors/:actorId', async (req, res, next) => {
+  try {
+    const { actorId } = req.params;
+    if (!Number.isInteger(+actorId)) {
+      throw new ClientError(400, `Non-integer actorId: ${actorId}`);
+    }
+    const sql = `
+      select * from "actors"
+      where "actorId" = $1;
+    `;
+    const params = [actorId];
+    const result = await db.query(sql, params);
+    const actor = result.rows[0];
+    if (!actor) throw new ClientError(404, `actor ${actorId} not found`);
+    res.json(actor);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/actors', async (req, res, next) => {
+  try {
+    const { firstName, lastName } = req.body;
+    if (!firstName || !lastName) {
+      throw new ClientError(400, 'firstName and lastName required');
+    }
+    const sql = `
+    insert into "actors" ("firstName", "lastName")
+    values ($1, $2)
+    returning *`;
+    const params = [firstName, lastName];
+    const result = await db.query(sql, params);
+    const actor = result.rows[0];
+    res.status(201).json(actor);
+  } catch (error) {
+    next(error);
+  }
+});
+
+app.put('/api/actors/:actorId', async (req, res, next) => {
+  try {
+    const { firstName, lastName } = req.body;
+    const { actorId } = req.params;
+    if (!firstName || !lastName) {
+      throw new ClientError(
+        400,
+        'missing required fields of firstName and lastName'
+      );
+    }
+    const checkSql = `
+    select * from "actors"
+    where "actorId" = $1`;
+    const checkParams = [actorId];
+    const checkResult = await db.query(checkSql, checkParams);
+    const actor = checkResult.rows[0];
+    if (!actor) {
+      throw new ClientError(400, `actor ${actorId} not found`);
+    }
+    const updateSql = `
+    update "actors"
+    set "firstName" = $1, "lastName" = $2
+    where "actorId" = $3
+    returning *`;
+    const updateParams = [firstName, lastName, actorId];
+    const updateResult = await db.query(updateSql, updateParams);
+    const updateActor = updateResult.rows[0];
+    res.status(200).json(updateActor);
+  } catch (error) {
+    next(error);
+  }
+});
+
+app.delete('/api/actors/:actorId', async (req, res, next) => {
+  try {
+    const { actorId } = req.params;
+
+    const deleteSql = `
+    delete from "actors"
+    where "actorId" = $1
+    returning *;`;
+
+    const deleteParams = [actorId];
+    const deleteResult = await db.query(deleteSql, deleteParams);
+    const deleteActor = deleteResult.rows[0];
+    if (!deleteActor) {
+      throw new ClientError(404, `actorId ${actorId} not found`);
+    }
+    res.status(204).send();
```

Comment:
Rather than using `send()`, you can use `res.sendStatus(204)`, so we update both the status code, and send a response back